### PR TITLE
[TACACS+]: Add configuration support for TACACS+

### PR DESCRIPTION
**Pull request by liuqu**
* Add config and show commands for TACACS+
* Add hostcfgd to listen configDB for TACACS+ and AAA, modify the
  pam configuration for Authentication in host.

  Signed-off-by: chenchen.qcc@alibaba-inc.com

### CODE REVIEW

**Comment by liuqu** [comment]
Move hostcfgd to sonic-buildimage [[TACACS+]: Add configDB enforcer for TACACS+ #1214](https://github.com/Azure/sonic-buildimage/pull/1214)

**Comment by lguohan** [comment]
can you resolve the conflict? also can you list all command list provided in your commit message.

**Comment by lguohan** [comment]
@jleveque to review.

**Comment by liuqu** [comment]
@lguohan The conflict has been resolved. The command list is amended to the last commit message.

File: config/aaa.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    if not val:
+        return False
+    try:
+        netaddr.IPAddress(str(val))
+    except:
+        return False
+    return True
+
+
+def set_entry(table, entry, data):
```
Comment (lguohan) [inline config/aaa.py]: the semantic of set_entry is changed in this https://github.com/Azure/sonic-py-swsssdk/commit/b7861cc4ae872eb91fa5c5c114eff43cd9e44ade.

if you still want to use the existing semantic, you can use mod_entry added in this https://github.com/Azure/sonic-py-swsssdk/pull/25

Comment (liuqu) [reply config/aaa.py]: Cool, it's better to use mod_entry.

File: config/aaa.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    """User authentication"""
+    pass
+aaa.add_command(authentication)
+
+
+# cmd: aaa authentication failthrough
+@click.command()
+@click.argument('option', type=click.Choice(["enable", "disable", "default"]))
+def failthrough(option):
+    """Allow AAA fail-through [enable | disable | default]"""
```
Comment (lguohan) [inline config/aaa.py]: what is the default value here? enable/disable?

Comment (lguohan) [reply config/aaa.py]:
```
gulv@str-s6000-on-4:~$ show aaa
AAA authentication login tacacs+,local
```
when it is set to default, what is the actual value? enable or disable? same question for failthrough

Comment (liuqu) [reply config/aaa.py]: The default value means there is no value for this configuration in configdb. For AAA authentication, the default is local authentication. For AAA fail-through, the default is enable.
I will add all the default value in the show command.

File: show/main.py [hunk 3]
```diff
@@ -778,6 +778,42 @@ def services():
...
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    data = config_db.get_table('AAA')
+    output = ''
+    if data != {}:
+        for row in data:
+            entry = data[row]
+            for key in entry:
+                output += ('AAA %s %s %s\n' % (row, key, str(entry[key])))
+
```
Comment (lguohan) [inline show/main.py]: can we print out the default values for fallback and fallthough?

Comment (liuqu) [reply show/main.py]: OK, I'll change it.

**Pull request merged**